[vscode]Leaking java debug process

### Description
When start debugging using vscode java-debug extension, the debug process will remain in OS process list and never be killed by theia process if the page is reloaded.
If I don't kill the debug process manually and start a new debug session, the debug console will show the exception of "Address already in use".

### Reproduction Steps
* install jdk and maven
```
apt-get install openjdk-11-jdk
apt-get install maven
```
* git clone https://github.com/theia-ide/theia.git && cd theia
* Delete theia java and java-debug extensions, for I only want to use debug function provided by vscode java-debug extension
```
sed -i '/theia\/java/d' examples/browser/package.json
mkdir plugins
wget https://marketplace.visualstudio.com/_apis/public/gallery/publishers/redhat/vsextensions/java/0.44.0/vspackage -O plugins/redhat.java-0.44.0.vsix
wget https://marketplace.visualstudio.com/_apis/public/gallery/publishers/vscjava/vsextensions/vscode-java-debug/0.15.0/vspackage -O plugins/vscjava.vscode-java-debug-0.15.0.vsix
```
* yarn
* cd examples/browser && yarn start
* Open a spring java project (you can git clone https://github.com/spring-projects/spring-petclinic ) and press F5 (for now you may need to press twice) to start debugging session.
* Click browser reload button and press F5 (Start Debugging button), the debug console will show error logs.
```
org.apache.catalina.util.LifecycleBase : Failed to start component [Connector[HTTP/1.1-8080]]
org.apache.catalina.LifecycleException: Protocol handler start failed
Caused by: java.net.BindException: Address already in use
```
**OS and Theia version:**
OS: Ubuntu 18.04
Theia version: latest master version

**Diagnostics:** 
1. click **Start Debugging** button and reload the page
![leaking-debug-process-p1](https://user-images.githubusercontent.com/7686800/58449976-7e18fb80-813f-11e9-8de4-bfa443844e0d.gif)
2. click **Start Debugging** button again, but failed to start the debug session
![leaking-debug-process-p2](https://user-images.githubusercontent.com/7686800/58449983-83764600-813f-11e9-9eaa-eaf3766b303b.gif)
3. there are some errors in **Debug Console**
![leaking-debug-process-p3](https://user-images.githubusercontent.com/7686800/58449990-8b35ea80-813f-11e9-87db-c59ce7eb6505.gif)
4. leaking java debug process
![normal-debug-process](https://user-images.githubusercontent.com/7686800/58463026-af0c2700-8165-11e9-80c5-daa558c0c552.PNG)

--- a/packages/plugin-ext/package.json
+++ b/packages/plugin-ext/package.json
@@ -29,6 +29,7 @@
     "getmac": "^1.4.6",
     "jsonc-parser": "^2.0.2",
     "lodash.clonedeep": "^4.5.0",
+    "ps-tree": "^1.2.0",
     "request": "^2.82.0",
     "uuid": "^3.2.1",
     "vscode-debugprotocol": "^1.32.0",
@@ -70,7 +71,8 @@
   "devDependencies": {
     "@theia/ext-scripts": "^0.9.0",
     "@types/decompress": "^4.2.2",
-    "@types/lodash.clonedeep": "^4.5.3"
+    "@types/lodash.clonedeep": "^4.5.3",
+    "@types/ps-tree": "^1.1.0"
   },
   "nyc": {
     "extends": "../../configs/nyc.json"

--- a/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts
+++ b/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts
@@ -25,6 +25,7 @@ import { RPCProtocolImpl } from '../../api/rpc-protocol';
 import { MAIN_RPC_CONTEXT } from '../../api/plugin-api';
 import { HostedPluginCliContribution } from './hosted-plugin-cli-contribution';
 import { HostedPluginProcessesCache } from './hosted-plugin-processes-cache';
+import * as psTree from 'ps-tree';
 
 export interface IPCConnectionOptions {
     readonly serverName: string;
@@ -129,9 +130,17 @@ export class HostedPluginProcess implements ServerPluginRunner {
         const hostedPluginManager = rpc.getProxy(MAIN_RPC_CONTEXT.HOSTED_PLUGIN_MANAGER_EXT);
         hostedPluginManager.$stopPlugin('').then(() => {
             emitter.dispose();
-            cp.kill();
+            this.killProcessTree(cp.pid);
         });
+    }
 
+    private killProcessTree(parentPid: number): void {
+        psTree(parentPid, (err: Error, childProcesses: Array<psTree.PS>) => {
+            childProcesses.forEach((p: psTree.PS) => {
+                process.kill(parseInt(p.PID));
+            });
+            process.kill(parentPid);
+        });
     }
 
     public runPluginServer(): void {

